code/datums/position_point_vector.dm

```diff
@@ -89,143 +87,90 @@
 		_pixel_y = A.pixel_y
 	initialize_location(_x, _y, _z, _pixel_x, _pixel_y)
 
-/datum/point/proc/initialize_location(tile_x, tile_y, tile_z, p_x = 0, p_y = 0)
+/datum/point/proc/initialize_location(tile_x, tile_y, tile_z, p_x, p_y)
 	if(!isnull(tile_x))
-		x = ((tile_x - 1) * ICON_SIZE_X) + ICON_SIZE_X * 0.5 + p_x + 1
+		x = tile_x
 	if(!isnull(tile_y))
-		y = ((tile_y - 1) * ICON_SIZE_Y) + ICON_SIZE_Y * 0.5 + p_y + 1
+		y = tile_y
 	if(!isnull(tile_z))
 		z = tile_z
+	if(!isnull(p_x))
+		var/x_offset = SIGN(p_x) * FLOOR(abs(p_x) / ICON_SIZE_X, 1)
```

Comment:
it is concerning that sign breaks behavior, not relevant here but thought, should we update FLOOR to respect sign (or make a new macro to allow differing behavior?)